dbms/src/Interpreters/ExpressionAnalyzer.cpp

```diff
@@ -122,7 +122,7 @@ void ExpressionAnalyzer::analyzeAggregation()
         ASTPtr array_join_expression_list = select_query->array_join_expression_list(is_array_join_left);
         if (array_join_expression_list)
         {
-            getRootActions(array_join_expression_list, true, temp_actions);
+            getRootActions(array_join_expression_list, true, temp_actions, false, true);
```

Comment:
Too much flags. It's better to make separate function getRootActionsNoMakeSet().
It's also not clean why it's possible to disable making set only in these 3 getRootActions() calls, leave in others and keep optimisation alive.

Reply:
Ok, I've updated as advised and added some explanation comments